.github/workflows/build-test-windows.yml

```diff
@@ -2,11 +2,23 @@
 name: build-and-test-windows
 on:
   push:
-    branches: [ main, master ]
+    branches:
+      - master
+
+    paths-ignore:
+      - '.github/**'
+      - '!.github/workflows/build.yml'
+      - '!.github/workflows/build-test-windows.yml'
```

Comment:
Instead of this can't we just put `- '!.github/workflows/**'`?

Reply:
For ```nightly-build, integration-test or close-stalled```, I believe it does not affect to the main branch until it has been triggered by ```dispatch``` or ```schedule``` though. However, for build and build-test-windows it would affect immediately when we do a merge so I don't see any reasons to include ```nightly-build, integration-test or close-stalled```

Reply:
We will have more workflows in the future and a `/**` would future proof us instead of having to add absolute paths here each time a new file is added.

Reply:
Reasonable for me at that solution but still, I find it weird to trigger the workflow is unrelated to branch master which will be triggered later  though

Reply:
If this is an issue, we could leverage the file names for the workflows to differentiate between what needs to be checked and what doesn't. Maybe something like
```yaml
  paths-ignore:
    - '.github/**'
    - '!.github/workflows/build-*.yml'
```
This would continue to ignore `nightly-build`, `integration-test`, and `close-stalled`

Reply:
Make sense. Fixed it in the newest commit and nice catch though